### PR TITLE
Updated unittests for template matching

### LINKED ISSUE #58: normaliseUnderMask can sometimes return NaN
test_NormTest.py fails every other time with ValueError.
I tried to debug and found that the issue is actually happening in basic/correlation.py around line 223, `stdValueUnderMask` becomes a NaN. I am not sure why that happens, more investigating is needed.

For now I just ignore this exception with a try-catch (see commit 5c5f8c07b32a4f018c539baa418a5a2ae00a2625).

### LINKED ISSUE #94: template matching GPU unittest should be run in default
Template matching GPU unittest is now present in the E2Etests folder, but should be moved up to the default tests folder so that it is always executed. It could get a decorator that specifies not to run in docker environment, or when there is no gpu available.

### PR DESCRIPTION
**Pull request by McHaillet**
Apparently I did something wrong with the commits. After the pull request for the voltools problem, I continued working on that same branch. I guess that messed some things up... I dont know

Anyway, this is what I updated for this pull request, which mainly revolves around a better template matching unittest:
    
- template matching now has a proper test that runs the algorithm. However, setting the gpu environment for the unittest is not straightforward. I eventually made a separate file (template_match_test.py) that is called via os.system, and is not found by 'discover'. The gpu test needs a fresh environment to load gpu code correctly. See more info in LocalizationTest.
- LocalizationTest had a suite that did not run properly. Now it does with unittest discover.
- NormTest had an unfinished test... I updated it
- CPU template matching had some issues after the pytom.lib import
- I changed the package_data in setup.py so that the angle lists are actually copied to the conda environment. Luckily, the classes that loaded the angle list did so based on the relative path of pytom (`pytom.__path__`). So, the lists are found correctly.
- If GPU cannot be loaded during unittesting, the interpolation-test will now show a BIG warning to notify the developper! (based on previous discussion :) )

btw this closes #94 and closes #58

### CODE REVIEW

**Review by sroet — COMMENTED** [review]
No worries, I can luckily select which commits to review in the review screen.

I had one question, LGTM otherwise

File: pytom/localization/extractPeaks.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def templateMatchingGPU(volume, reference, rotations, scoreFnc=None, mask=None,
     from pytom.gpu.gpuStructures import TemplateMatchingGPU
     from pytom.tools.calcFactors import calc_fast_gpu_dimensions
     import numpy as np
-    from pytom.gpu.initialize import xp
+    from pytom.gpu.initialize import xp, device
```
[the remark this answers is not in the document]
Comment (McHaillet) [reply pytom/localization/extractPeaks.py:200]: I think this is okay. In `pytom.gpu.initialize`, numpy or cupy is assigned to a `global xp` depending on the 'PYTOM_GPU' env variable. The problem occurs when `pytom.gpu.initialize` is imported before 'PYTOM_GPU' is set in `os.environ`. So, importing numpy as np before wont change that.

**Comment by McHaillet** [comment]
I am actually not a big fan of this PYTOM_GPU environment variable. As this complicated test execution shows, you can only run GPU code after setting this variable at the start--which is know taken care of in the pytom executable. It is pretty cumbersome.

I think it would be better to use the cupy adviced method (https://docs.cupy.dev/en/stable/user_guide/basic.html#how-to-write-cpu-gpu-agnostic-code). You can use `cp.get_array_module()` to load backend from an ndarray. Its very easy for a simple image filter function, e.g.

```python
import cupy as cp

def low_pass_filter(image):
    xp = cp.get_array_module(image)
    # create_low_pass outputs np/cp array depending on 
    low_pass = create_low_pass_filter(image.shape, backend=xp)
    return xp.fft.ifftn(xp.fft.fftn(image) * low_pass)
```

**Comment by sroet** [comment]
> I am actually not a big fan of this PYTOM_GPU environment variable. As this complicated test execution shows, you can only run GPU code after setting this variable at the start--which is know taken care of in the pytom executable. It is pretty cumbersome.
>
> I think it would be better to use the cupy adviced method (https://docs.cupy.dev/en/stable/user_guide/basic.html#how-to-write-cpu-gpu-agnostic-code). You can use cp.get_array_module() to load backend from an ndarray. Its very easy for a simple image filter function, e.g.

I think that would be a nice simplification, do you mind making an issue out of it?

**Review by sroet — APPROVED** [review]
All comments have been handled. LGTM, merging

**Comment by McHaillet** [comment]
Yep, I'll make an issue!

**Pull request merged**